tests/drivers/gpio/gpio_basic_api/boards/sim.conf

```diff
@@ -0,0 +1,4 @@
+# Copyright (c) 2020 Friedt Professional Engineering Services, Inc
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_GPIO_SIM=y
```

Comment:
If I understand what you want you are trying to do*1: This file should be named tests/drivers/gpio/gpio_basic_api/boards/native_posix.conf

*1: Are you are trying to enable this config when native_posix is the target?